nixos/modules/hardware/temp-throttle.nix

```diff
@@ -0,0 +1,64 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.temp-throttle;
+  configFile = with builtins;
+    "MAX_TEMP=" + toString cfg.max_temp + "\n" +
+    (if cfg.interval  != null then "INTERVAL="  + toString cfg.interval + "\n" else "") +
+    (if cfg.temp_file != null then "TEMP_FILE=" + cfg.interval          + "\n" else "") +
+    (if cfg.core      != null then "CORE="      + toString cfg.core     + "\n" else "") +
+    (if cfg.log_file  != null then "LOG_FILE="  + cfg.interval          + "\n" else "")
+  ;
+in
+{
+  options = with lib; with lib.types; {
```

Comment:
Don't use `with lib;`, especially in a nested way.
See #208242.

Reply:
I assume the same holds true for the service module. Is keeping `with lib.types` okay?

Reply:
> Is keeping with `lib.types` okay?

No, better to avoid it.
See rationale in https://nix.dev/guides/best-practices#with-scopes.